datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -257,7 +257,56 @@ impl PhysicalExpr for BinaryExpr {
     fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
         use arrow::compute::kernels::numeric::*;
 
+        #[inline]
+        fn check_short_circuit(arg: &ColumnarValue, op: &Operator) -> bool {
+            let data_type = arg.data_type();
+            if data_type == DataType::Boolean {
+                if *op == Operator::And {
+                    match arg {
+                        ColumnarValue::Array(array) => {
+                            if let Ok(array) = as_boolean_array(&array) {
+                                if array.true_count() == 0 {
```

Comment:
Nit: We can return inline `return array.true_count() == 0` here. Same for other places below